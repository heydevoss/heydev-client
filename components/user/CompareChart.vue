<template>
  <vue-no-ssr>
    <div>
      <div>
        <chartjs-bar
          class="chart"
          :beginzero="true"
          :datasets="datasets"
          :labels="labels"
          :option="options"
        />
      </div>
    </div>
  </vue-no-ssr>
</template>

<script>
export default {
  props: {
    prContributor: Number,
    issuesContributor: Number,
    commitsContributor: Number,
    prOrganization: Number,
    issuesOrganization: Number,
    commitsOrganization: Number
  },
  data() {
    return {
      beginZero: true,
      labels: ['Pull Requests', 'Issues', 'Commits'],
      datasets: [
        {
          backgroundColor: '#4ECDC4',
          data: [
            this.prContributor,
            this.issuesContributor,
            this.commitsContributor
          ],
          label: 'Contributor'
        },
        {
          backgroundColor: '#FF6B6B',
          data: [
            this.prOrganization,
            this.issuesOrganization,
            this.commitsOrganization
          ],
          label: 'Organization'
        }
      ],
      options: {
        maintainAspectRatio: true,
        legend: {
          position: 'right'
        }
      }
    }
  }
}
</script>

<style scoped>
.chart {
  width: 500px;
  height: 500px;
}

@media screen and (max-width: 820px) {
  .chart {
    width: 300px;
    height: 300px;
  }
}
</style>
