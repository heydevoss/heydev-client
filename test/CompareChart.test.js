import { mount } from '@vue/test-utils'
import CompareChart from '@/components/user/CompareChart.vue'

describe('CompareChart', () => {
  const wrapper = mount(CompareChart)

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('renders the table ', () => {
    expect(wrapper.html()).toContain('<chartjs-bar')
  })
})
