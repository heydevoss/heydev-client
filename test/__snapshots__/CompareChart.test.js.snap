// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompareChart renders correctly 1`] = `
<vue-no-ssr>
  <div>
    <div>
      <chartjs-bar
        beginzero="true"
        class="chart"
        datasets="[object Object],[object Object]"
        labels="Pull Requests,Issues,Commits"
        option="[object Object]"
      />
    </div>
  </div>
</vue-no-ssr>
`;
